### DISCUSSION TITLE
Just-in-Time compiler with calc function does not apply proper style

### DISCUSSION THREAD

**Discussion by cookieMonsterDev** [tailwindlabs/tailwindcss#12723] [Help]
Recently worked on a new project with nextjs, got a strange behaviour when tried to use calc inside JIT compiler like:  h-[calc(100svh - 200px)], not sure why be the style does not work((( Maybe someone has some similar?

```tsx
"use client";
import React from "react";

const ProjectsError = () => {
  return (
    <div className="w-full h-[calc(100svh - 50svh)] grid place-content-center border p-2">
      <h1 className="text-destructive">
        Something went wrong during loading projects!
      </h1>
    </div>
  );
};

export default ProjectsError;
```

![image](https://github.com/tailwindlabs/tailwindcss/assets/72692356/2d4035c2-1dce-4ba6-9bf4-520d30cb4717)

**Comment by wongjn** [comment]
Spaces delimit class names so you cannot have spaces in arbitrary values. For `calc()` values, Tailwind should add spaces in for you, so instead of `h-[calc(100svh - 50svh)]`, do `h-[calc(100svh-50svh)]`. [Relevant documentation](https://tailwindcss.com/docs/content#using-spaces-and-underscores).
